File: ANCIEN FRONT/services/stock/transfers.service.ts
import {fetcher} from "../utils/fetch-utils.service.ts";
import {BASE_API_CIBLE_HREF} from "../../constants.ts";
import TransfersModel from "../../core/domain/stock/TransfersModel.ts";


export function getAllTransfers(): Promise<TransfersModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-transfers`, requestOptions);
}

export function createTransfers(object: TransfersModel): Promise<TransfersModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/transfers`, requestOptions);
}

File: ANCIEN BACK/cible/domain/stock/services/transfers_service.py
from cible.domain.stock.interface.i_transfers_repository import ITransfersRepository
from cible.domain.stock.models.transfers_bean import TransfersBean


def get_all_transfers(repository: ITransfersRepository) -> list[TransfersBean]:
    return repository.get_all_transfers()


def create_transfers(
    repository: ITransfersRepository, transfers_bean: TransfersBean
) -> list[TransfersBean]:
    return repository.create_transfers(transfers_bean)
